### PR TITLE
Update default blockwise token to avoid DataFrame column name clash

### PR DESCRIPTION
**Pull request by jrbourbeau**
Our current default blockwise token prefix ("_") can interact poorly with DataFrames that have column names like `_0`, `_1`, etc. The PR proposes we update our default token prefix to something more unique ("\_\_default__") that's less likely to clash with user inputs. 

cc @jcrist for review if you get a moment

Closes https://github.com/dask/dask/issues/6540

- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: dask/blockwise.py [hunk 1]
```diff
@@ -37,7 +37,10 @@ def index_subs(ind, substitution):
         return tuple([substitution.get(c, c) for c in ind])
 
 
-def blockwise_token(i, prefix="_"):
+blockwise_default_prefix = "__default__"
```
Comment (goodwanghan) [inline dask/blockwise.py:40]: is it possible to change to __dask_default__?

**Comment by jsignell** [comment]
This seems like a very reasonable low-impact solution to me. And I think `__dask_default__` or `__dask_blockwise__` would be fine names.

**Comment by jsignell** [comment]
@jrbourbeau did you want to get this one in today?

**Comment by jrbourbeau** [comment]
Let's hold off on this one to rock the boat before releasing. But after 2.30.0 is out, this should be good to go

**Comment by jsignell** [comment]
@jrbourbeau mind if I pick this up and finish it off?

**Comment by jrbourbeau** [comment]
Yeah, feel free to 👍 I'm curious if @rjzamora's recently blockwise HLG work will be impacted by this or not

File: dask/blockwise.py (hunk 1 above)
Comment (rjzamora) [reply dask/blockwise.py:40]: Perhaps a label that makes it more obvious that it is an abstract "blockwise" label would be even better (like `"__blockwise__"`).  Though,  I certainly dont feel strongly about this.

**Comment by rjzamora** [comment]
>I'm curious if @rjzamora's recently blockwise HLG work will be impacted by this or not

I certainly hope changing the default string doesn't break anything, but anything that breaks shouldn't be depending on the "_" default anyway :)

**Comment by jsignell** [comment]
Tests pass for me locally. My plan is to merge when green, unless there are any last comments.

**Review by jrbourbeau — COMMENTED** [review]
Thanks @jsignell!

**Pull request merged**